src/test/java/com/github/wovnio/wovnjava/ApiTest.java

```diff
@@ -76,13 +83,66 @@ private static String testTranslate(byte[] apiServerResponse, String encoding, i
                                      "\"version\":\"" + Settings.VERSION + "\"," +
                                      "\"debug_mode\":\"false\"," +
                                      "\"translate_canonical_tag\":\"true\"," +
+                                     "\"user_agent\":\"WOVN Browser\"," +
                                      "\"body\":\"\u003Chtml\u003Emuch content\u003C\\/html\u003E\"}";
 
         assertEquals(expectedRequestBody, apiRequestBody);
 
         return result;
     }
 
+    public void testGetApiUrl() throws ApiException, IOException, ProtocolException, ConfigurationError, ApiNoPageDataException, NoSuchAlgorithmException {
+      Settings settings = TestUtil.makeSettings(new HashMap<String, String>() {{
+          put("projectToken", "token0");
+          put("defaultLang", "en");
+          put("supportedLangs", "en,ja,fr");
+          put("urlPattern", "path");
+      }});
+      UrlLanguagePatternHandler urlLanguagePatternHandler = UrlLanguagePatternHandlerFactory.create(settings);
+
+      HttpServletRequest request = MockHttpServletRequest.create("https://example.com/ja/somepage/", "WOVN Browser");
+      ResponseHeaders responseHeaders = mockResponseHeaders();
+
+      Headers headers = new Headers(request, settings, urlLanguagePatternHandler);
+      RequestOptions requestOptions = new RequestOptions(settings, request);
+
+      Api api = new Api(settings, headers, requestOptions, responseHeaders, Clock.systemDefaultZone());
+      String expected_url = "https://wovn.global.ssl.fastly.net/v0/translation?cache_key=%28token%3Dtoken0%26settings_hash%3D5C76D283B72EEA106B91CE18830A2EE4%26body_hash%3DD41D8CD98F00B204E9800998ECF8427E%26path%3D%2Fsomepage%2F%26lang%3Den%26version%3Dwovnjava_" + Settings.VERSION + "%29";
+      assertEquals(expected_url, api.getApiUrl("en", "").toString());
+    }
+
+    public void testGetApiUrlWithSearchEngineBot() throws ApiException, IOException, ProtocolException, ConfigurationError, ApiNoPageDataException, NoSuchAlgorithmException {
```

Comment:
i would like to see some mention of the timestamp in the test name. Like 
`testGetApiUrlWithSearchEngineBot__cachedEvery20Minutes`